From e28c6c531c0f5e2646dd654ed9e6e0a7ef2af88e Mon Sep 17 00:00:00 2001
From: Daniel Kiper <daniel.kiper@oracle.com>
Date: Thu, 25 Feb 2021 18:35:01 +0100
Subject: util/grub-install: Fix NULL pointer dereferences

Two grub_device_open() calls does not have associated NULL checks
for returned values. Fix that and appease the Coverity.

Fixes: CID 314583

Signed-off-by: Daniel Kiper <daniel.kiper@oracle.com>
Reviewed-by: Javier Martinez Canillas <javierm@redhat.com>

Patch-Name: 2021-02-security/053-util-grub-install-Fix-NULL-pointer-dereferences.patch
---
 util/grub-install.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/util/grub-install.c b/util/grub-install.c
index 90b0705fa..6afa457db 100644
--- a/util/grub-install.c
+++ b/util/grub-install.c
@@ -1916,6 +1916,8 @@ main (int argc, char *argv[])
 	  fill_core_services (core_services);
 
 	  ins_dev = grub_device_open (install_drive);
+	  if (ins_dev == NULL)
+	    grub_util_error ("%s", grub_errmsg);
 
 	  bless (ins_dev, core_services, 0);
 
@@ -2016,6 +2018,8 @@ main (int argc, char *argv[])
 	  fill_core_services(core_services);
 
 	  ins_dev = grub_device_open (install_drive);
+	  if (ins_dev == NULL)
+	    grub_util_error ("%s", grub_errmsg);
 
 	  bless (ins_dev, boot_efi, 1);
 	  if (!removable && update_nvram)
